OBJC_INTEROP.md

```diff
@@ -0,0 +1,124 @@
+# _Kotlin/Native_ interoperability with Swift/Objective-C
+
+This documents covers some details of Kotlin/Native interoperability with
+Swift/Objective-C.
+
+## Usage
+
+Kotlin/Native provides bidirectional interoperability with Objective-C.
+Objective-C frameworks and libraries can be used in Kotlin code if
+properly attached to the build (system frameworks are attached by default).
+See e.g. "Interop libraries" in
+[Gradle plugin documentation](GRADLE_PLUGIN.md#building-artifacts).
+Swift library can be used in Kotlin code if its API is exported to Objective-C
+with `@objc`.
+
+Kotlin module can be used in Swift/Objective-C code if compiled into a
+[framework](GRADLE_PLUGIN.md#framework). See [calculator sample](samples/calculator)
+as an example.
+
+## Mappings
+
+The table below shows how Kotlin concepts are mapped to Swift/Objective-C and vice versa.
+
+| Kotlin | Swift | Objective-C | Notes |
+| ------ | ----- |------------ | ----- |
+| `class` | `class` | `@interface` | [note](#name-translation) |
+| `interface` | `protocol` | `@protocol` | |
+| `constructor` | Initializer | Initializer | |
+| Property | Property | Property | [note](#top-level-functions-and-properties) |
+| Method | Method | Method | [note](#top-level-functions-and-properties) |
+| `@Throws` | `throws` | `error:(NSError**)error` | [note](#errors-and-exceptions) |
+| Extension | Extension | Category member | [note](#category-members) |
+| Companion object member <- | Class method or property | Class method or property |  |
+| `Singleton` | `Singleton()`  | `[Singleton singleton]` | |
+| Primitive type | Primitive type / `NSNumber` | | [note](#number) |
+| `String` | `String` | `NSString` | |
+| `String` | `NSMutableString` | `NSMutableString` | [note](#nsmutablestring) |
+| `List` | `Array` | `NSArray` | |
+| `MutableList` | `NSMutableArray` | `NSMutableArray` | |
+| `Set` | `Set` | `NSSet` | |
+| `MutableSet` | `NSMutableSet` | `NSMutableSet` | [note](#mutable-collections) |
+| `Map` | `Dictionary` | `NSDictionary` | |
+| `MutableMap` | `NSMutableDictionary` | `NSMutableDictionary` | [note](#mutable-collections) |
+| Function type | Function type | Block pointer type | |
+
+### Name translation
+
+Objective-C classes are imported into Kotlin with their original names.
+Protocols are imported as interfaces with `Protocol` name suffix,
+i.e. `@protocol Foo` -> `interface FooProtocol`.
+These classes and interfaces are placed into package [specified in build configuration](#usage)
+(`platform.*` packages for preconfigured system frameworks).
+
+Names of Kotlin classes and interfaces are prefixed when imported to Swift/Objective-C.
+The prefix is derived from the framework name.
+
+### Top-level functions and properties
```

Comment:
Unit conversion paragraph?

Reply:
Yes, it makes sense. What do you think, should it be related to function types mapping or to `Unit` type mapping?

Reply:
Maybe function types.